### ISSUE TITLE
[MNT]: Specify ordering in file in gallery folder

### ISSUE THREAD

**Issue by story645** [matplotlib/matplotlib#25032]
> Since we wrote this extension to do the ordering, there's no reason we can't change it to read from `README.txt` in each directory. We could have it look for something that _looks_ like a comment to reST so it does nothing in the sphinx-gallery-processed reST without having to do anything extra.
_Originally posted by @QuLogic in https://github.com/matplotlib/matplotlib/issues/24746#issuecomment-1396482343_

I think it's a relatively good first issue following #25028 because each readme can be parsed into its respective gallery list. Steps involved are 
1. figure out the reST comment notation that allows for easy specification of ordering + a consistent sentinel for placing unsorted files
2. write a parser that translates the set of comments into a list that preserves the order (including sentinel)
3. parse each file into its respective list in https://github.com/matplotlib/matplotlib/blob/be7ea76c49f19ce48958de82a82b3951d9b7f0d2/doc/sphinxext/gallery_order.py#L48
4. add documentation on doing this ordering to https://github.com/matplotlib/matplotlib/blob/main/doc/devel/documenting_mpl.rst?plain=1#L933

Medium difficulty because some familiarity with sphinx would be helpful, but not strictly necessary. I think a medium to advanced version of this would be implementing a `.. gallery-toc` [custom directive](https://www.sphinx-doc.org/en/master/development/tutorials/index.html#extension-tutorials-index)

 And yes, down the line we should maybe spin a maintain docs page off from the writing docs page. 

An alternative to using the readme is using a special purpose 'gallery_order' file https://github.com/matplotlib/matplotlib/issues/24746#issuecomment-1399116931

**Comment by jklymak** [comment]
I was going to open a discussion with sphinx-gallery about this as well.  It seems like something they should be able to handle.

**Comment by QuLogic** [comment]
I looked into this yesterday; it's not particularly difficult to implement, and I did about half of it. The subsection order is pretty easy to do as each one is independent, but the problem with the higher section order is that we currently have a mix with some special things appearing after the unsorted ones. I did not fully investigate if the ordering class gets access to all the directories, so there's a little bit of extra work to handle the bookkeeping of the split there.

**Comment by Higgs32584** [comment]
I want to do this issue, but I am a little bit confused by this part:
write a parser that translates the set of comments into a list that preserves the order (including sentinel)

Are you pulling the text from each README.md or reordering the directory? Thank you!

**Comment by Kruthikakatta** [comment]
I’d like to work on this issue.

**Comment by timhoffm** [comment]
@story645 Is this still needed / worth the effort?

Given the constrains of sphinx-gallery due to their ordering logic, https://github.com/matplotlib/matplotlib/blob/main/doc/sphinxext/gallery_order.py isn't too bad.

We have a declarative order specification in there. Given that changes to the gallery order are are, an order file in the gallery doesn't seem to bring much additional benefit. And I'm reluctant to add more complex logic that we'd have to maintain.

**Comment by story645** [comment]
>  Given that changes to the gallery order are are, an order file in the gallery doesn't seem to bring much additional benefit. 

Yeah, I'm fine closing this and maybe repopening if there's a change to the docs where it'd be really useful to order by file and can't just use the sphinx minigallery command to do it.

**Comment by parthivdholiya77-coder** [comment]
Hi, I am new to open source and would like to work on this issue.
Can you please guide me on where to start?

**Comment by timhoffm** [comment]
This is closed. No action planned.
